src/main/scala/nl/hugo/redbook/ch7/Par.scala

```diff
@@ -0,0 +1,114 @@
+package nl.hugo.redbook.ch7
+
+import java.util.concurrent._
+import language.implicitConversions
+
+object Par {
+  type Par[A] = ExecutorService => Future[A]
+
+  def run[A](s: ExecutorService)(a: Par[A]): Future[A] = a(s)
```

Comment:
I guess it has to do that this chapter is actually about API design (so creating the abstraction), and not about doing parallel computations. Somewhere halfway this chapter, this concept is lost.